x-pack/legacy/plugins/epm/public/screens/add_data_source/index.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useState, useEffect } from 'react';
+import { ICON_TYPES, EuiPageHeader, EuiTitle, EuiFlexGroup, EuiIcon, EuiPanel } from '@elastic/eui';
+import styled from 'styled-components';
+import { NavButtonBack } from '../../components/nav_button_back';
+import { useLinks } from '../../hooks';
+import { PackageInfo } from '../../../common/types';
+import { getPackageInfoByKey } from '../../data';
+import { AddDataSourceSteps } from './add_data_source_steps';
+
+export interface AddDataSourceProps {
+  pkgkey: string;
+}
+
+// TODO: change to percentages?
+const SIDEBAR_WIDTH = '168px';
+
+const PageContainer = styled.div`
```

Comment:
Isn't this the same container with, column number, and gutter size as the detail page? Why can we use that layout or duplicate it here?

Reply:
The detail page's column is too wide at 192px when it should be around 168.  This layout is also simpler than the Detail page not needing the 2 EuiPage components to split the layout and have the gray background effect in the header area. Also the layout for this page is different because if you look at the design there needs to be more space on the right side than the left of the layout (not centered) and it doesn't have a 3rd column.  Henry and I also discussed that EuiPage does not work well with our layout which is more custom.  My idea is to have a file with some custom variable widths that we can use in both areas and consolidate some components we can share like the side nav column.  So to answer your question yes we can consolidate some things but I wanted to push it to another issue for now.  
https://github.com/elastic/kibana/issues/52321

Reply:
I think we can still use the same approach for the outer content columns, but we can discuss / implement in that issue